src/core/mavsdk_impl.cpp

```diff
@@ -394,45 +453,84 @@ bool MavsdkImpl::does_system_exist(uint8_t system_id)
     return false;
 }
 
-void MavsdkImpl::notify_on_discover(const uint64_t uuid)
+void MavsdkImpl::notify_on_discover(const uint8_t system_id, const uint8_t component_id)
 {
+    // TODO kalyan - where is this called?
     if (_on_discover_callback != nullptr) {
-        _on_discover_callback(uuid);
+        _on_discover_callback(system_id, component_id);
     }
 }
 
-void MavsdkImpl::notify_on_timeout(const uint64_t uuid)
+void MavsdkImpl::notify_on_timeout(const uint8_t system_id, const uint8_t component_id)
 {
-    LogDebug() << "Lost " << uuid;
+    // TODO kalyan - where is this called?
+    LogDebug() << "Lost " << system_id << " " << component_id;
     if (_on_timeout_callback != nullptr) {
-        _on_timeout_callback(uuid);
+        _on_timeout_callback(system_id, component_id);
     }
 }
 
 void MavsdkImpl::register_on_discover(const Mavsdk::event_callback_t callback)
 {
+    // TODO kalyan - where is this called?
     std::lock_guard<std::recursive_mutex> lock(_systems_mutex);
 
     if (callback) {
-        for (auto const& connected_system : _systems) {
-            // Ignore dummy system with system ID 0.
-            if (connected_system.first == 0) {
-                continue;
-            }
-            // Ignore system if UUID is not initialized yet.
-            if (connected_system.second->get_uuid() == 0) {
-                continue;
-            }
-            callback(connected_system.second->get_uuid());
+        for (auto const& connected_node : _nodes) {
+            callback(connected_node.second->get_system_id(), connected_node.second->get_component_id());
         }
+        //for (auto const& connected_system : _systems) {
+            //// Ignore dummy system with system ID 0.
+            //if (connected_system.first == 0) {
+                //continue;
+            //}
+            //// TODO where will this be called? We shouldn't have a scenario where the uuid isn't init
+            //// Ignore system if UUID is not initialized yet.
+            //if (connected_system.second->get_uuid() == 0) {
+                //continue;
+            //}
+            //callback(connected_system.second->get_uuid());
+        //}
     }
 
     _on_discover_callback = callback;
 }
 
 void MavsdkImpl::register_on_timeout(const Mavsdk::event_callback_t callback)
 {
+    // TODO kalyan - where is this called?
     _on_timeout_callback = callback;
 }
 
+void MavsdkImpl::system_thread()
+{
+    dl_time_t last_time{};
+
+    while (!_should_exit) {
+        if (_time.elapsed_since_s(last_time) >= MavsdkImpl::_HEARTBEAT_SEND_INTERVAL_S) {
+            send_heartbeat();
+        }
+        last_time = _time.steady_time();
+
+        std::this_thread::sleep_for(std::chrono::milliseconds(50)); // in case we want to do anything else in this thread
+    }
+}
+
+void MavsdkImpl::send_heartbeat()
+{
+    std::cout << "Sending heartbeat" << std::endl;
```

Comment:
You should always use `LogDebug()` instead of `std::cout`.

Reply:
Done!